content/spotworkers/workers.md

```diff
@@ -0,0 +1,88 @@
+---
+title: "Add EC2 Workers - On-Demand and Spot"
+date: 2018-08-07T11:05:19-07:00
+weight: 10
+draft: false
+---
+
+We have our EKS Cluster and worker nodes already, but we need some Spot Instances configured as workers. We also need a Node Labeling strategy to identify which instances are Spot and which are on-demand so that we can make more intelligent scheduling decisions. We will use [AWS CloudFormation](https://aws.amazon.com/cloudformation/) to launch new worker
+nodes that will connect to the EKS cluster.
+
+This template will create a single ASG that leverages the latest feature to mix multiple instance types and purchase as a single K8s nodegroup. Check out this blog: [New – EC2 Auto Scaling Groups With Multiple Instance Types & Purchase Options](https://aws.amazon.com/tw/blogs/aws/new-ec2-auto-scaling-groups-with-multiple-instance-types-purchase-options/) for details.
+
+#### Retrieve the Worker Role name
+
+First, we will need to collect the Role Name that is in use with our EKS worker nodes
+
+```bash
+INSTANCE_PROFILE_PREFIX=$(aws cloudformation describe-stacks --stack-name eksctl-eksworkshop-eksctl-nodegroup-0 | jq -r '.Stacks[].Outputs[].ExportName' | sed 's/:.*//')
```

Comment:
the node group stack name doesn't seem to be eksctl-eksworkshop-eksctl-nodegroup-0 anymore. It is eksctl-eksworkshop-eksctl-nodegroup-ng-xxxxxxxx now

Reply:
@vjsikha - I just ran through this again today and it worked as expected. The worker node group created by eksctl matches this pattern described above so the steps to capture the Role Name are correct.

The CFN template that we use to create the spot workers does use a slightly different naming scheme.

Let's chat about this if you still have concerns

Reply:
Added new export logic for node group/role name. Exporting to bash profile so that it is available to any workshop module that needs to reference it